mailbox/api/src/test/java/org/apache/james/mailbox/events/EventDeadLettersContract.java

```diff
@@ -406,5 +406,30 @@ default void groupsWithFailedEventsShouldReturnEmptyWhenNoStored() {
 
             assertThat(eventDeadLetters.groupsWithFailedEvents().toStream()).isEmpty();
         }
+
+        @Test
+        default void groupsWithFailedEventShouldReturnRemainStoredGroupsAfterRemoveOneEvent() {
```

Comment:
IMO we have no need to remove the GROUP B here

That's a remark @mbaechler  did to us during the grooming.

IMO you can relaxe both of these tests.

Reference to this remark: RMK-1